log4j-compress/pom.xml

```diff
@@ -0,0 +1,186 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to you under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.logging.log4j</groupId>
+    <artifactId>log4j</artifactId>
+    <version>${revision}</version>
+    <relativePath>../log4j-parent</relativePath>
+  </parent>
+
+  <artifactId>log4j-compress</artifactId>
```

Comment:
*Thinking out loudly:* Shouldn't we rather rename this to `log4j-compress-commons`, in case we want to support another 3rd party library in the future? :thinking:

Reply:
Maybe it would be prudent, but I would prefer 3rd party compression libraries to integrate with Commons Compress instead of us. I did add a custom `CompressorStreamProvider` to prove that 3rd party extensions to Commons Compress can be used by this module.

While writing this PR I also looked if we can integrate [CLP](https://github.com/y-scope/clp) through this mechanism. The answer is no: CLP would require us to compress each log event separately, it is not useful to compress already written log files (at least without parsing them again).